src/parentheses/Solution.java

```diff
@@ -0,0 +1,50 @@
+package parentheses;
+
+import java.util.Stack;
+
+public class Solution {
+    public boolean isValid(String parentheses) {
+
+        char[] arrayParentheses = parentheses.toCharArray();
+        Stack<Character> stack = new Stack<>();
+        boolean allOk = true;
+        for (int i = 0; i < arrayParentheses.length; i++) {
+            char character = arrayParentheses[i];
+            boolean isOpencharacter = character == '[' || character == '{' || character == '(';
+
+            if (isOpencharacter) {
+                stack.push(character);
+                System.out.println("stack: " + stack);
+            } else {
+                switch (character) {
+                    case ')':
+                        allOk = checkCloseParentheses(stack, '(');
+                        break;
+                    case '}':
+                        allOk = checkCloseParentheses(stack, '{');
+                        break;
+                    case ']':
+                        allOk = checkCloseParentheses(stack, '[');
+                        break;
+                    default:
+                        throw new IllegalStateException("Unexpected value: " + arrayParentheses[i]);
+                }
+            }
+        }
+        return allOk;
+    }
+
+    private boolean checkCloseParentheses(Stack<Character> stack, char openCharacter) {
```

Comment:
`checkCloseParentheses` is a good name but it could be better. One of the things to improve about this name is that it is not providing information about what it will return.

A better name would be `areParenthesesBalanced`. If it returns `true` then the parenthesis are balanced, if it returns false, they are not.

Reply:
ok, rename "checkCloseParentheses" by "areParenthesesBalanced". 
thanks for the observation, this makes the code easier to read.